src/coffeescript.coffee

```diff
@@ -100,12 +100,16 @@ exports.compile = compile = withPrettyErrors (code, options = {}) ->
   )
 
   # Check for import or export; if found, force bare mode.
-  unless options.bare? and options.bare is yes
+  sourceType = options.sourceType ? 'unambiguous'
```

Comment:
This is based on the definition of `sourceType` in babel. In both babel `sourceType: 'unambiguous'` and in CoffeeScript, there's behavior based on the existence of `import`/`export` statements. In the future this could also fail properly when `import` is used in scripts but that may fall out of the scope of coffeescript.

Reply:
FYI the original code here removes the [“safety wrapper”](https://coffeescript.org/#lexical-scope) `(function() { ... }).call(this);` when the source code contains `import` or `export`, since per spec `import` and `export` must be at the top level and therefore such a wrapper would always break ESM code.

Reply:
Another problem with the IFFE is that `.call(this)` in a module wouldn't work. So even without `import`/`export` it would break when the source type is module. :)